### ISSUE TITLE
Validators throw on optional input arguments with validation

### ISSUE THREAD

**Issue by sergeyshaykhullin** [benmccallum/fairybread#32]
To reproduce add optional input argument and register validator for it

Now you are checking `argument != null`, but you need to check `value != null`

```
Unhandled exception. System.ArgumentNullException: Cannot pass null model to Validate. (Parameter 'instanceToValidate')
   at FluentValidation.Internal.Extensions.Guard(Object obj, String message, String paramName) in C:\Projects\FluentValidation\src
\FluentValidation\Internal\Extensions.cs:line 36
   at FluentValidation.AbstractValidator`1.EnsureInstanceNotNull(Object instanceToValidate) in C:\Projects\FluentValidation\src\Fl
uentValidation\AbstractValidator.cs:line 353
   at FluentValidation.AbstractValidator`1.Validate(ValidationContext`1 context) in C:\Projects\FluentValidation\src\FluentValidat
ion\AbstractValidator.cs:line 114
   at FluentValidation.AbstractValidator`1.Validate(T instance) in C:\Projects\FluentValidation\src\FluentValidation\AbstractValid
ator.cs:line 62

```

Missed null check https://github.com/benmccallum/fairybread/blob/a265376da51fe9483694794cd6c3ae3ce779b97c/src/FairyBread/InputValidationMiddleware.cs#L45

**Comment by benmccallum** [comment]
Hmm, that's interesting... If you have the time, feel free to submit a PR; else I'll look to fix this this week/weekend.

**Comment by benmccallum** [comment]
Fixed in 3.0.1.

Thanks again for reporting :)
